src/liquid/components/ld-notification/ld-notification.tsx

```diff
@@ -0,0 +1,190 @@
+import '../../components' // type definitions for type checks and intelliSense
+import { Component, h, Host, Listen, Prop, State, Watch } from '@stencil/core'
+
+type Notification = {
+  type: 'info' | 'warn' | 'alert'
+  content: string
+  timeout?: number
+}
+
+const DEFAULT_NOTIFICATION_TIMEOUT = 6000
+const FADE_TRANSITION_DURATION = 200
+
+@Component({
+  tag: 'ld-notification',
+  styleUrl: 'ld-notification.css',
+  shadow: false,
+})
+export class LdNotification {
+  /**
+   * Notification placement within the screen.
+   */
+  @Prop() placement: 'top' | 'bottom' = 'top'
+
+  @State() queue: Notification[] = []
+  @State() queueDismissed: Notification[] = []
+
+  @State() dismissTimeout: NodeJS.Timeout | null
+
+  @State() fadeoutTimeouts: (NodeJS.Timeout | null)[] = []
+
+  @State() currentNotification?: Notification
+
+  @Watch('currentNotification')
+  updateDismissTimeout() {
+    clearTimeout(this.dismissTimeout)
+
+    if (!this.currentNotification) return
+    if (this.currentNotification.type === 'alert') return
+    if (this.currentNotification.timeout === 0) return
+
+    this.dismissTimeout = setTimeout(() => {
+      this.handleNotificationDismiss()
+    }, this.currentNotification.timeout || DEFAULT_NOTIFICATION_TIMEOUT)
+  }
+
+  @Listen('ldNotification', {
```

Comment:
Not sure what you mean by using `connectedCallback` and `disconnectedCallback` as event-replacement. These hooks are component life cycle methods and have nothing to do with the actual notification events. The only thing they can be used for in this context is to add and remove event listeners... Anyway, I decided for the use of just one instance of the component for three reasons:
1. The container of all notifications should be placed outside other containers in order to circumvent z-index issues.
2. It makes it easier to handle only one single queue, instead of consolidating multiple queues into one.
3. For the developer it is actually super easy to trigger events. There is no need for lifting state up; just fire the event and forget.

I have changed the event name to `ldNotificationAdd`.

Reply:
Totally fine with that. Points 1 + 2 would actually also apply to my suggested alternate approach. If you want to, I can explain to you in a call. But lets just leave it as is.